forest/daemon/src/daemon.rs

```diff
@@ -73,6 +73,16 @@ pub(super) async fn start(opts: CliOpts, config: Config) -> anyhow::Result<Db> {
     }
 
     set_sigint_handler();
+
+    let mem_stats_tracker = if opts.track_peak_rss {
+        Some(MemStatsTracker::default())
+    } else {
+        None
+    };
+    if let Some(mem_stats_tracker) = &mem_stats_tracker {
+        mem_stats_tracker.run_async();
```

Comment:
Can this background task be added to the services?

Reply:
Yes, but that will make its usage a bit complicated for other tools, e.g. statediff

Reply:
How about exposing the async loop and then letting the caller spawn a task? Or maybe just use `std::thread::spawn`.

Reply:
Fixed

Reply:
Thanks! Now I won't accidentally run the function from a non-async context. :)